fabric-content-registries-v0/src/main/java/net/fabricmc/fabric/impl/content/registry/FabricAbstractBlock.java

```diff
@@ -0,0 +1,105 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.content.registry;
+
+import org.jetbrains.annotations.Nullable;
+
+import net.minecraft.block.AbstractBlock;
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.ai.pathing.PathNodeType;
+
+/**
+ * General-purpose Fabric-provided extensions for {@link AbstractBlock} subclasses.
+ *
+ * <p>Note: This interface is automatically implemented on the {@link AbstractBlock} class
+ * via Mixin and interface injection.
+ *
+ * <p>Note to maintainers: Functions should only be added to this interface if they are general-purpose enough,
+ * to be evaluated on a case-by-case basis. Otherwise, they are better suited for more specialized APIs.
+ */
+public interface FabricAbstractBlock {
+	/**
+	 * Gets the {@link PathNodeType} for the block.
+	 *
+	 * <p>This returns the {@link PathNodeType} if the block is a direct target in an entity path.
+	 *
+	 * <p>For example, in this case, for a cactus-like block you should specify {@link PathNodeType#DAMAGE_CACTUS},
+	 * so entities will avoid to pass through or above the block.
+	 *
+	 * <p>Tip: Return null if you want to use vanilla default behavior.
+	 *
+	 * @param state Current block state.
+	 * @return the {@link PathNodeType} specified for the block.
+	 */
+	@Nullable
+	default PathNodeType getPathNodeType(BlockState state) {
+		return null;
+	}
+
+	/**
+	 * Gets the {@link PathNodeType} for the block if is in a closest neighbor position
+	 * that the entity will find in his path.
+	 *
+	 * <p>This returns the {@link PathNodeType} if the block is a direct target in an entity path.
+	 *
+	 * <p>For example, in this case, for a cactus-like block you should specify {@link PathNodeType#DANGER_CACTUS},
+	 * so entities will avoid to get close to the block.
+	 *
+	 * <p>Tip: Return null if you want to use vanilla default behavior.
+	 *
+	 * @param state Current block state.
+	 * @return the {@link PathNodeType} specified for the block.
+	 */
+	@Nullable
+	default PathNodeType getNeighborPathNodeType(BlockState state) {
+		return null;
+	}
+
+	/**
+	 * General-purpose Fabric-provided extensions for {@link AbstractBlock.AbstractBlockState} subclasses.
+	 *
+	 * <p>Note: This interface is automatically implemented on the {@link AbstractBlock.AbstractBlockState} class
+	 * via Mixin and interface injection.
+	 *
+	 * <p>Note to maintainers: Functions should only be added to this interface if they are general-purpose enough,
+	 * to be evaluated on a case-by-case basis. Otherwise, they are better suited for more specialized APIs.
+	 */
+	interface FabricAbstractBlockState {
+		/**
+		 * Gets the {@link PathNodeType} for the block state.
+		 *
+		 * @return the {@link PathNodeType} specified for the block state.
```

Comment:
Clarify that this is for direct targets of a path

```suggestion
		 * Gets the {@link PathNodeType} for the block state if it is a direct target in an entity path.
		 *
		 * @return the {@link PathNodeType} specified for the block state.
```